pkg/gateway/kubernetes.go

```diff
@@ -66,8 +66,11 @@ func (asm *apiServiceMapper) ServiceMap() (*ServiceMap, error) {
 		for _, port := range service.Spec.Ports {
 			if port.Port != ingServicePort || port.Protocol != kapi.ProtocolTCP {
 				continue
+			} else {
+				svc.ListenPort = port.NodePort
```

Comment:
It's unusual to me that this loop would continue past the first match. I believe the break is more explicit about how this works in practice.